sympy/core/add.py

```diff
@@ -576,6 +580,8 @@ def _eval_is_positive(self):
             v = _monotonic_sign(a)
             if v is not None:
                 s = v + c
+                if s == self:
+                    return None
                 if s.is_positive and a.is_nonnegative:
```

Comment:
I really am still quite baffled by what this logic does, but I guess if `s == self` it should skip the whole monotonic_sign stuff, but instead of returning None, continue to the logic below it.